src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java

```diff
@@ -38,7 +39,15 @@
  * introduction of new types in an enumeration type. Note that
  * the compiler requires switch expressions to be exhaustive,
  * so this check does not enforce default branches on
- * such expressions.
+ * such expressions. Also, switch statements that use pattern or null
+ * labels are checked by the compiler for completeness (exhaustiveness), so
```

Comment:
switch statement - Check do validation
switch expression - Check does no validation.
`.... Also, switch expression that use pattern or null labels are checked by the compiler for completeness/exhaustiveness (example: it is compile error if any enum value is missed), so`

please fix other places statement -> expression.

Reply:
> switch expression - Check does no validation.

if this is true, we probably can state simple like this.

Reply:
>switch expression... we probably can state simple like this. At the begining, and keep all other details following.

Good idea, I agree.

> switch statement - Check do validation

> please fix other places where you have statement -> expression.

This is not accurate; the check will not validate switch statements that use pattern or null labels (as described in documentation in this PR), since the compiler does this. In fact, as described in the javadocs in my new example, it would cause a compiler error if a default clause were added, in the case of totality. Example:

```

➜  src cat Test.java
public class Test {

    void method(String s) {
        switch (s) {
            case Object o:
                System.out.println("Object!");
            default:
                System.out.println("default!");
        }
    }
}
➜  src javac --enable-preview --release 17 Test.java
Test.java:7: error: switch has both a total pattern and a default label
            default:
                   ^
Note: Test.java uses preview features of Java SE 17.
Note: Recompile with -Xlint:preview for details.
1 error


```

Compiler enforcing exhaustiveness for switch statement with pattern label:

```
➜  src cat Test.java                                
import java.util.ArrayList;
import java.util.List;

public class Test {

    void method(List<String> l) {
        switch (l) {
            case ArrayList ar:
                System.out.println("arraylist!");
        }
    }
}
➜  src javac --enable-preview --release 17 Test.java
Test.java:7: error: the switch statement does not cover all possible input values
        switch (l) {
        ^
Note: Test.java uses preview features of Java SE 17.
Note: Recompile with -Xlint:preview for details.
1 error


```

Similarly, also consider:

```
➜  src cat Test.java                                
import java.util.ArrayList;
import java.util.List;

public class Test {

    void method(List<String> l) {
        switch (l) {
            case null:
                System.out.println("arraylist!");
        }
    }
}
➜  src javac --enable-preview --release 17 Test.java
Test.java:7: error: the switch statement does not cover all possible input values
        switch (l) {
        ^
Note: Test.java uses preview features of Java SE 17.
Note: Recompile with -Xlint:preview for details.
1 error

```

Reply:
Ok, I made language a lot more clear and went into details in the examples.